spirv_cross.cpp

```diff
@@ -2378,8 +2378,10 @@ SPIREntryPoint &Compiler::get_entry_point()
 bool Compiler::interface_variable_exists_in_entry_point(uint32_t id) const
 {
 	auto &var = get<SPIRVariable>(id);
-	if (var.storage != StorageClassInput && var.storage != StorageClassOutput)
-		SPIRV_CROSS_THROW("Only Input and Output variables are part of a shader linking interface.");
+	if (var.storage != StorageClassInput && var.storage != StorageClassOutput &&
+	    var.storage != StorageClassUniformConstant)
```

Comment:
Because `add_interface_block(StorageClassUniformConstant)` is called to create uniform structure. Then in the `add_interface_block` method, `interface_variable_exists_in_entry_point` is called and thus compiler error thrown.